src/jit/flowgraph.cpp

```diff
@@ -13490,13 +13490,15 @@ bool Compiler::fgOptimizeSwitchBranches(BasicBlock* block)
         GenTree* switchVal = switchTree->gtOp.gtOp1;
         noway_assert(genActualTypeIsIntOrI(switchVal->TypeGet()));
 
+#ifndef LEGACY_BACKEND
```

Comment:
I guess that is OK,  This section of code was previously not compiling at all for x86 I take it.

In general I am not a fan of having conditionally compilation for the GT_NODEs, 
 if GT_JMPTABLE is defined for one target it should be defined for all targets.

We don't have to use or fully implement it for all targets, but it is crazy to conditionally define the enum value.